src/util/mod.rs

```diff
@@ -52,6 +52,28 @@ macro_rules! error_type {
     );
 }
 
+macro_rules! derive_header {
```

Comment:
Do you think there's value in passing a `&'static str` as part of the name, so that external users of `headers` could create their own typed headers more easily?

Reply:
The stuff used in the header isn't public at all. It's mostly to avoid a bunch of internal boilerplate.